traffic_ops/traffic_ops_golang/profile/profiles.go

```diff
@@ -439,9 +439,9 @@ func Read(w http.ResponseWriter, r *http.Request) {
 	profileInterfaces := []interface{}{}
 
 	for _, p := range profileList {
-		// Attach Parameters if the 'param' parameter is sent
-		if _, ok := inf.Params["param"]; ok {
-			p.Parameters, err = ReadParameters(inf.Tx, inf.User, p.ID, inf.User.Can("PARAMETER-SECURE:READ"))
+		// Attach Parameters if the 'id' parameter is sent
+		if _, ok := inf.Params["id"]; ok {
+			p.Parameters, err = ReadParameters(inf.Tx, inf.User, p.ID, inf.User.Can("PARAMETER:SECURE-READ"))
```

Comment:
does this mean you need to have the PARAMETER:SECURE-READ permission to perform a profile compare?

Reply:
no, this was an incorrect check originally which wasn't listsing the params associated with a profile.